scripts/Modules/TTK_Lib.py

```diff
@@ -48,31 +55,40 @@ def encodeTrickInput(input):
 # Reads binary data in-memory for the specified section
 def readRawRKGData(playerType: PlayerType, inputType: ControllerInputType) -> list:
     retList = []
-    curAddr = 0x0
-    endAddr = 0x0
-    addresses = []
-    endAddrOffset = 0
     
     # Determine memory region to access
     if (playerType == PlayerType.PLAYER):
-        addresses = classes.getInputStorageAddresses()
-        endAddrOffset = 0x276C
+        # This assumes player is index 0
+        # For now let's assert this is a player
+        race_config = RaceConfig()
+        race_scenario = RaceConfigScenario(addr=race_config.race_scenario())
+        race_config_player = RaceConfigPlayer(addr=race_scenario.player(playerIdx=0))
+        assert(race_config_player.type() == RaceConfigPlayerType.REAL_LOCAL)
+
+        ghost_writer = GhostWriter(addr=PlayerInput.ghost_writer())
+        stream_addr = ghost_writer.button_stream(inputType.value)
+        button_stream = GhostButtonsStream(addr=stream_addr)
+        address = button_stream.buffer()
+        endAddrOffset = button_stream.size()
     else:
-        addresses = classes.getGhostAddresses()
-        endAddrOffset = memory.read_u32(classes.getGhostAddressLengthPointer()[2])
+        # TODO: Ghost is index=1 if you are racing a ghost, but if you watch a replay
+        # for example, the ghost is index 0. We want to support this scenario, so
+        # we'll need to determine how to set controllerIdx appropriately.
```

Comment:
The answer is to evaluate the current gamemode in RaceConfig.

Reply:
See https://github.com/TASLabz/mkw-scripts/issues/42